### PR TITLE
Add GNU standards and C18 standard for querying compiler

### PR DESCRIPTION
**Pull request by michelleangela**
Adds the following GNU standards to query compiler to get #defines for GNU standards.
(issue https://github.com/microsoft/vscode-cpptools/issues/2782)

Note: if GNU standard is specified in IntelliSense configuration, IntelliSense will emulate the equivalent C or C++ standard version.

GNU C++ standard
- gnu++20
- gnu++17
- gnu++14
- gnu++11
- gnu++03
- gnu++98

GNU C standard
- gnu18
- gnu11
- gnu99
- gnu89

Also, add C18 standard.

### CODE REVIEW

**Comment by sean-mcmanus** [comment]
It seems like we should use c18 instead of c17 https://en.wikipedia.org/wiki/C18_(C_standard_revision)

The backend should probably handle both on the command line (compilerArgs) though.

**Comment by michelleangela** [comment]
I don't see `c18` for Clang at https://clang.llvm.org/docs/CommandGuide/clang.html

`c18` is supported by GNU https://gcc.gnu.org/onlinedocs/gcc/C-Dialect-Options.html

**Comment by sean-mcmanus** [comment]
> I don't see c18 for Clang at https://clang.llvm.org/docs/CommandGuide/clang.html
> c18 is supported by GNU https://gcc.gnu.org/onlinedocs/gcc/C-Dialect-Options.html

c18 with clang works for me when I try it via https://godbolt.org/ . The docs might be out of date.

File: Extension/c_cpp_properties.schema.json [hunk 1]
```diff
@@ -26,17 +26,22 @@
                         }
                     },
                     "cStandard": {
-                        "description": "Version of the C language standard to use for IntelliSense.",
+                        "description": "Version of the C language standard to use for IntelliSense. Note: GNU standards are only used to query the set compiler to get GNU defines and IntelliSense will emulate the equivalent C standard version.",
```
Comment (michelleangela) [inline Extension/c_cpp_properties.schema.json]: Is description clear? please advice if changes are needed.

Comment (sean-mcmanus) [reply Extension/c_cpp_properties.schema.json]: Yeah, it should be "defines, and" though. Also, "query the set compiler" might be more easy to understand as "query the compilerPath`.

**Review by sean-mcmanus — APPROVED** [review]

**Pull request merged**